### PR TITLE
Supporting RequestRateLimiterGatewayFilterFactory

### PR DESCRIPTION
**Pull request by xiaoy00**
Please answer these questions before submitting pull request

- Why submit this pull request?
- [1] Bug fix
- [ ] New feature provided
- [ ] Improve performance

- Related issues
no
___
### Bug fix
- Bug description.
when i use the RequestRateLimiterGatewayFilterFactory for limit.
then ContextManager.capture() can not work in NettyRoutingFilterInterceptor.class
because Thread is changed from 'reactor-http-nio' to 'lettuce-nioEventLoop' 

- How to fix?
get ContextSnapshot on FilteringWebHandlerInterceptor and set it into SWTransmitter.
then NettyRoutingFilterInterceptor can use ContextSnapshot to   ContextManager.continued(swTransmitter.getSnapshot());
and i have to create a new LocalSpan for bring ContextManager.continued() worked 
___
### New feature or improvement
- Describe the details and related test reports.

### CODE REVIEW

**Comment by wu-sheng** [comment]
@xiaoy00 Could you provide demo codes about this?

**Comment by zhaoyuguang** [comment]
The current [gateway test demo](https://github.com/SkyAPMTest/agent-auto-integration-testcases/tree/master/gateway-scenario), it would be better to enrich this case here.

**Comment by xiaoy00** [comment]
> The current [gateway test demo](https://github.com/SkyAPMTest/agent-auto-integration-testcases/tree/master/gateway-scenario), it would be better to enrich this case here.

I have already submitted the case on  gateway test demo, please have a look

**Comment by zhaoyuguang** [comment]
Please modify the code according to code style and modify testcase so that test cases can work.

**Comment by xiaoy00** [comment]
> Hi @xiaoy00 , Please modify the code according to code style and modify testcase so that test cases can work.
you said about gateway-2.1.x-plugin or  [gateway test demo ](https://github.com/SkyAPMTest/agent-auto-integration-testcases/tree/master/gateway-scenario) ?

**Comment by zhaoyuguang** [comment]
Fix gateway-2.1.x-plugin according to code style.
And make [gateway test demo ](https://github.com/SkyAPMTest/agent-auto-integration-testcases/tree/master/gateway-scenario) can work.

**Comment by xiaoy00** [comment]
> Fix gateway-2.1.x-plugin according to code style.
> And make [gateway test demo ](https://github.com/SkyAPMTest/agent-auto-integration-testcases/tree/master/gateway-scenario) can work.
check style is ok on my idea. 
demo can work too, by the way , its need redis to work. 
i dont know what's the error . and i showed you on QQ.
u can give me some error infos.

**Comment by wu-sheng** [comment]
You should verify style by maven command. It is what we used on CI. Also, click details link, we have the logs.

**Comment by wu-sheng** [comment]
The referred demo is integratuon test, you need to read the plugin tests document and change the codes, then make tests passed.

In open source, we need to make sure cases are checked in any further changes. But not just your local tests.
Please understand, we need to make sure the whole community good, and make sure all upstream changes are safe and could be reverified always.
You need to spend mire time on tests than code changes.

File: apm-sniffer/optional-plugins/optional-spring-plugins/optional-spring-cloud/gateway-2.1.x-plugin/pom.xml [hunk 1]
```diff
@@ -34,6 +34,7 @@
 
 <properties>
     <spring-cloud-starter-gateway.version>2.1.1.RELEASE</spring-cloud-starter-gateway.version>
+    <compiler.version>1.8</compiler.version>
```
Comment (wu-sheng) [inline apm-sniffer/optional-plugins/optional-spring-plugins/optional-spring-cloud/gateway-2.1.x-plugin/pom.xml]: Are you using jdk8 style api? Why do you change this?

Comment (xiaoy00) [reply apm-sniffer/optional-plugins/optional-spring-plugins/optional-spring-cloud/gateway-2.1.x-plugin/pom.xml]: yes, i use Mono.doFinally()

Comment (xiaoy00) [reply apm-sniffer/optional-plugins/optional-spring-plugins/optional-spring-cloud/gateway-2.1.x-plugin/pom.xml]: sorry i have some changes please look at the last commit

**Comment by wu-sheng** [comment]
#3603 merged. Please continue.

**Comment by xiaoy00** [comment]
> #3603 merged. Please continue.
roger!

**Comment by wu-sheng** [comment]
Resolve the conflicts, please.

**Comment by zhaoyuguang** [comment]
run agent-plugin-test-3

**Pull request merged**